### PR TITLE
 Laboratory work #3, Karina Melkonyan - 23FPL2

### PR DESCRIPTION
**Pull request by melkern**

### CODE REVIEW

File: lab_2_retrieval_w_bm25/settings.json [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-    "target_score": 0
+    "target_score": 8
```
Comment (Vasilisa-Blyudova) [inline lab_2_retrieval_w_bm25/settings.json:2]: revert

File: lab_3_ann_retriever/main.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 Vector search with text retrieving
 """
-
+import re
```
Comment (Vasilisa-Blyudova) [inline lab_3_ann_retriever/main.py:6]: import after `# pylint`

File: lab_3_ann_retriever/main.py [hunk 3]
```diff
@@ -255,7 +271,7 @@ def index_documents(self, documents: list[str]) -> bool:
         """
 
     def retrieve_relevant_documents(
-        self, query: str, n_neighbours: int
+            self, query: str, n_neighbours: int
```
Comment (Vasilisa-Blyudova) [inline lab_3_ann_retriever/main.py:274]: revert

File: lab_3_ann_retriever/main.py [hunk 4]
```diff
@@ -306,7 +322,7 @@ def retrieve_vectorized(self, query_vector: Vector) -> str | None:
         """
 
     def _calculate_knn(
-        self, query_vector: Vector, document_vectors: list[Vector], n_neighbours: int
+            self, query_vector: Vector, document_vectors: list[Vector], n_neighbours: int
```
Comment (Vasilisa-Blyudova) [inline lab_3_ann_retriever/main.py:325]: revert

File: lab_3_ann_retriever/main.py [hunk 5]
```diff
@@ -366,11 +382,11 @@ class Node(NodeLike):
     right_node: NodeLike | None
 
     def __init__(
-        self,
-        vector: Vector = (),
-        payload: int = -1,
-        left_node: NodeLike | None = None,
-        right_node: NodeLike | None = None,
+            self,
```
Comment (Vasilisa-Blyudova) [inline lab_3_ann_retriever/main.py:385]: .

File: lab_3_ann_retriever/main.py [hunk 6]
```diff
@@ -527,7 +543,7 @@ def index_documents(self, documents: list[str]) -> bool:
         """
 
     def retrieve_relevant_documents(
-        self, query: str, n_neighbours: int = 1
+            self, query: str, n_neighbours: int = 1
```
Comment (Vasilisa-Blyudova) [inline lab_3_ann_retriever/main.py:546]: .

**Review by Vasilisa-Blyudova — CHANGES_REQUESTED** [review]

File: lab_3_ann_retriever/main.py (hunk 2 above)
Comment (melkern) [reply lab_3_ann_retriever/main.py:6]: done

File: lab_3_ann_retriever/main.py (hunk 3 above)
Comment (melkern) [reply lab_3_ann_retriever/main.py:274]: done

File: lab_3_ann_retriever/main.py (hunk 4 above)
Comment (melkern) [reply lab_3_ann_retriever/main.py:325]: done

File: lab_3_ann_retriever/main.py (hunk 6 above)
Comment (melkern) [reply lab_3_ann_retriever/main.py:546]: done

File: lab_3_ann_retriever/main.py [hunk 7]
```diff
@@ -104,6 +112,10 @@ def tokenize(self, text: str) -> list[str] | None:
 
         In case of corrupt input arguments, None is returned.
         """
+        if not isinstance(text, str):
+            return None
+        text = re.findall(r'[a-zа-яё]+', text.lower())
```
Comment (artyomtugaryov) [inline lab_3_ann_retriever/main.py:117]: hm... Can you explain it?

File: lab_3_ann_retriever/main.py [hunk 8]
```diff
@@ -304,6 +388,16 @@ def retrieve_vectorized(self, query_vector: Vector) -> str | None:
...
         """
+        if not isinstance(query_vector, tuple) or not query_vector:
+            return None
+        if len(query_vector) != len(self._document_vectors[0]):
+            return None
+        retrieve_doc = self._calculate_knn(query_vector, self._document_vectors, 1)
+        if not isinstance(retrieve_doc, list) or not retrieve_doc:
+            return None
+        index, value = retrieve_doc[0]
+        document = self._documents[index]
```
Comment (artyomtugaryov) [inline lab_3_ann_retriever/main.py:399]: do not create the variable

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_3_ann_retriever/main.py (hunk 7 above)
Comment (melkern) [reply lab_3_ann_retriever/main.py:118]: I did the same way in the last lab and explained to my mentor

**Pull request closed without merging**